core/src/rpc.rs

```diff
@@ -839,6 +838,154 @@ impl JsonRpcRequestProcessor {
             inactive: inactive_stake,
         })
     }
+
+    pub fn get_token_account_balance(
+        &self,
+        pubkey: &Pubkey,
+        token_program_id: &Pubkey,
+        commitment: Option<CommitmentConfig>,
+    ) -> RpcResponse<u64> {
+        let bank = self.bank(commitment);
+        let balance = bank
+            .get_account(pubkey)
+            .and_then(|account| {
+                if token_program_id == &spl_token_id_v1_0() {
+                    parse_token(&account.data).ok()
+                } else {
+                    None
```

Comment:
It would be better to return an error to the user when the program id or TokenAccountType is unexpected so they can differentiate between an error state and a token accounts that's actually 0